src/content/docs/en/guides/integrations-guide/cloudflare.mdx

```diff
@@ -79,171 +79,139 @@ export default defineConfig({
+### imageService
 
 <p>
-
-**Type:** `'advanced' | 'directory'`<br />
-**Default:** `'advanced'`
+**Type:** `'passthrough' | 'cloudflare' | 'compile'`<br />
+**Default:** `'passthrough'`
 </p>
 
-This configuration option defines how your Astro project is deployed to Cloudflare Pages.
-
-* `advanced` mode picks up the `_worker.js` file in the `dist` folder
-* `directory` mode picks up the files in the `functions` folder, by default only one `[[path]].js` file is generated
-
-Switching to directory mode allows you to add additional files manually such as [Cloudflare Pages Plugins](https://developers.cloudflare.com/pages/platform/functions/plugins/), [Cloudflare Pages Middleware](https://developers.cloudflare.com/pages/platform/functions/middleware/) or custom functions using [Cloudflare Pages Functions Routing](https://developers.cloudflare.com/pages/platform/functions/routing/).
-
-```js title="astro.config.mjs" ins="mode: 'directory'"
-export default defineConfig({
-  adapter: cloudflare({ mode: 'directory' }),
-});
-```
+Determines which image service is used by the adapter. The adapter will default to `passthrough` mode when an incompatible image service is configured. Otherwise, it will use the globally configured image service:
 
-To compile a separate bundle for each page, set the `functionPerRoute` option in your Cloudflare adapter config. This option requires some manual maintenance of the `functions` folder. Files emitted by Astro will overwrite existing files with identical names in the `functions` folder, so you must choose unique file names for each file you manually add. Additionally, the adapter will never empty the `functions` folder of outdated files, so you must clean up the folder manually when you remove pages.
+* **`cloudflare`:** Uses the [Cloudflare Image Resizing](https://developers.cloudflare.com/images/image-resizing/) service.
+* **`passthrough`:** Uses the existing [`noop`](/en/guides/images/#configure-no-op-passthrough-service) service.
+* **`compile`:** Uses Astro's default service (sharp), but only on pre-rendered routes at build time. During SSR for pages rendered on-demand, all `astro:assets` features are disabled.
 
-```js title="astro.config.mjs" ins={7}
+```js title="astro.config.mjs" ins={6}
 import {defineConfig} from "astro/config";
 import cloudflare from '@astrojs/cloudflare';
 
 export default defineConfig({
   adapter: cloudflare({
-    mode: 'directory',
-    functionPerRoute: true
-  })
+     imageService: 'cloudflare'
+  }),
+  output: 'server'
 })
 ```
 
-This adapter doesn't support the [`edgeMiddleware`](/en/reference/adapter-reference/#edgemiddleware) option.
+### platformProxy
 
-### routes.strategy
+Determines whether and how the Cloudflare runtime is added to `astro dev`. It contains proxies (to local `workerd` bindings) and emulations of Cloudflare specific values, allowing the emulation of the runtime in the Node.js dev process. Read more about the [Cloudflare Runtime](#cloudflare-runtime).
```

Comment:
I know you do specify `local` but it might be worth to emphasise that the bindings work only locally and aren't related to the production thing

I'm just mentioning this as I've seen people being confused as to why they can't see their production data during local development

Reply:
Thanks for this heads up, we'll not add it to the initial version, but can adapt if we see to many confusions.